common/transport/amqp/src/amqp_cbs.ts

```diff
@@ -0,0 +1,323 @@
+import * as machina from 'machina';
+import * as amqp10 from 'amqp10';
+import * as uuid from 'uuid';
+
+import { errors } from 'azure-iot-common';
+import { AmqpMessage } from './amqp_message';
+import { SenderLink } from './sender_link';
+import { ReceiverLink } from './receiver_link';
+
+/**
+ * @interface  module:azure-iot-amqp-base.PutTokenOperation
+ * @classdesc  Describes a "put token" operation state: "put-token" operations are used to renew the CBS token and stay connected.
+ *
+ * @property   {Function}   putTokenCallback       The callback to be invoked on termination of the put token operation.
+ *                                                 This could be because the put token operation response was received
+ *                                                 from the service or because the put token operation times out.
+ * @property   {Number}     expirationTime         The number of seconds from the epoch, by which the put token operation will
+ *                                                 be expected to finish.
+ * @property   {Number}     correlationId          The put token operation was sent with a message id.  The response
+ *                                                 to the put token operation will contain this message id as the
+ *                                                 correlation id.  This id is a uuid.
+ *
+ */
+interface PutTokenOperation {
+  putTokenCallback: (err: Error | null, result?: any) => void;
+  expirationTime: number;
+  correlationId: string;
+}
+
+/**
+ * @class      module:azure-iot-amqp-base.PutTokenStatus
+ * @classdesc  Describes the state of the "put token" feature of the client that enables Claim-Based-Security authentication.
+ *
+ * @property   {Function}   outstandingPutTokens                This array will hold outstanding put token operations.  The array has
+ *                                                              a monotonically increasing time ordering.  This is effected by the nature
+ *                                                              of inserting new elements at the end of the array.  Note that elements may
+ *                                                              be removed from the array at any index.
+ * @property   {Number}     numberOfSecondsToTimeout            Currently a fixed value.  Could have a set option if we want to make this configurable.
+ * @property   {Number}     putTokenTimeOutExaminationInterval  While there are ANY put token operations outstanding a timer will be invoked every
+ *                                                              10 seconds to examine the outstandingPutTokens array for any put tokens that may have
+ *                                                              expired.
+ * @property   {Number}     timeoutTimer                        Timer used to trigger examination of the outstandingPutTokens array.
+ */
+class PutTokenStatus {
```

Comment:
>PutTokenStatus [](start = 6, length = 14)

PutToken is an unfortunate name.  It's not clear that Put is not a verb.  (I first thought this was a function that put the token status).  There's probably not a better way to name this, but I still have to ask.

Reply:
that's just code that moved, i really don't want to refactor that at the same time

---
In reply to: [139535819](https://github.com/Azure/azure-iot-sdk-node/pull/83#discussion_r139535819) [](ancestors = 139535819)